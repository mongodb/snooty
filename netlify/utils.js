import fs from 'fs';
import path from 'path';

/**
 * Parses build hook data from the expected temporary location
 * @returns {object | undefined}
 */
const parseBuildHookData = () => {
  // This file does not currently exist, but should be created on Netlify builds as part of `npm run build:netlify`.
  // The INCOMING_HOOK_BODY env var is not automatically passed along to functions, so we use a txt file to save it
  const relativeFilePath = '../build-hook.txt';
  const buildHookDataString = fs.readFileSync(path.resolve(__dirname, relativeFilePath), 'utf-8');
  if (!buildHookDataString) {
    console.log('No build hook data found.');
    return;
  }
  return JSON.parse(buildHookDataString);
};

export const constructResPayload = (event) => {
  const buildHookData = parseBuildHookData();
  const parsedEventBody = JSON.parse(event.body);
  // This is Netlify's default post-deployment payload. We include it with our custom data in case
  // we want to process any information
  const netlifyPayload = parsedEventBody.payload;
  return {
    netlifyPayload,
    ...buildHookData,
  };
};
