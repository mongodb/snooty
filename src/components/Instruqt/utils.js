export const getLabURL = (embedValue) => {
  return `https://play.instruqt.com/embed${embedValue}`;
};
