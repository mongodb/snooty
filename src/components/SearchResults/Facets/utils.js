/**
 *
 * @param {obj} facet - contains key and id properties as returned by /v2/status ie.
 * https://docs-search-transport.mongodb.com/v2/status
 *
 * @returns {str} blue | green
 */
export const getFacetTagVariant = (facet) => {
  if (facet.key === 'target_product') {
    return 'green';
  }
  return 'blue';
};
