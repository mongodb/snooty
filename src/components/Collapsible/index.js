import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { useLocation } from '@gatsbyjs/reach-router';
import Box from '@leafygreen-ui/box';
import Icon from '@leafygreen-ui/icon';
import IconButton from '@leafygreen-ui/icon-button';
import { cx } from '@leafygreen-ui/emotion';
import { Body } from '@leafygreen-ui/typography';
import { HeadingContextProvider } from '../../context/heading-context';
import { findAllNestedAttribute } from '../../utils/find-all-nested-attribute';
import { isBrowser } from '../../utils/is-browser';
import { reportAnalytics } from '../../utils/report-analytics';
import ComponentFactory from '../ComponentFactory';
import Heading from '../Heading';
import { collapsibleStyle, headerContainerStyle, headerStyle, iconStyle, innerContentStyle } from './styles';
import './styles.css';

const Collapsible = ({ nodeData: { children, options }, sectionDepth, ...rest }) => {
  const { id, heading, expanded, sub_heading: subHeading } = options;
  const { hash } = useLocation();

  // get a list of all ids in collapsible content
  // in order to set collapsible open, if any are found in url hash
  const childrenHashIds = useMemo(() => {
    return findAllNestedAttribute(children, 'id');
  }, [children]);

  const [open, setOpen] = useState(expanded ?? true);
  const headingNodeData = {
    id,
    children: [{ type: 'text', value: heading }],
  };

  const onIconClick = useCallback(() => {
    reportAnalytics('CollapsibleClicked', {
      action: open ? 'collapsed' : 'expanded',
      heading,
    });
    setOpen(!open);
  }, [heading, open]);

  // open the collapsible if the hash in URL is equal to collapsible heading id,
  // or if the collapsible's children has the id
  useEffect(() => {
    if (!isBrowser) {
      return;
    }
    const hashId = hash?.slice(1) ?? '';
    if (id === hashId || childrenHashIds.includes(hashId)) {
      return setOpen(true);
    }
  }, [childrenHashIds, hash, id]);

  const rendered = useRef(false);

  // on first open, scroll the child with the URL hash id into view
  useEffect(() => {
    if (!open) return;
    if (rendered.current) return;
    rendered.current = true;
    const hashId = hash?.slice(1) ?? '';
    if (childrenHashIds.includes(hashId)) {
      const child = document?.querySelector(`#${hashId}`);
      child && child.scrollIntoView({ block: 'start', inline: 'nearest', behavior: 'smooth' });
    }
  }, [childrenHashIds, hash, open]);

  return (
    <HeadingContextProvider ignoreNextHeading={true} heading={heading}>
      <Box className={cx('collapsible', collapsibleStyle)}>
        <Box className={cx(headerContainerStyle)}>
          <Box>
            {/* Adding 1 to reflect logic in parser, but want to show up as H2 for SEO reasons */}
            <Heading className={cx(headerStyle)} sectionDepth={sectionDepth + 1} as={2} nodeData={headingNodeData}>
              {heading}
            </Heading>
            <Body baseFontSize={13}>{subHeading}</Body>
          </Box>
          <IconButton className={iconStyle} aria-labelledby={'Expand the collapsed content'} onClick={onIconClick}>
            <Icon glyph={open ? 'ChevronDown' : 'ChevronRight'} />
          </IconButton>
        </Box>
        <Box className={cx(innerContentStyle(open))}>
          {children.map((c, i) => (
            <ComponentFactory nodeData={c} key={i} sectionDepth={sectionDepth} {...rest}></ComponentFactory>
          ))}
        </Box>
      </Box>
    </HeadingContextProvider>
  );
};

Collapsible.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object),
    options: PropTypes.shape({
      heading: PropTypes.string.isRequired,
      sub_heading: PropTypes.string,
      id: PropTypes.string,
    }),
  }),
};

export default Collapsible;
