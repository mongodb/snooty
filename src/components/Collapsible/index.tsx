import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useLocation } from '@gatsbyjs/reach-router';
import Box from '@leafygreen-ui/box';
import Icon from '@leafygreen-ui/icon';
import IconButton from '@leafygreen-ui/icon-button';
import { cx } from '@leafygreen-ui/emotion';
import { Body } from '@leafygreen-ui/typography';
import { HeadingContextProvider } from '../../context/heading-context';
import { findAllNestedAttribute } from '../../utils/find-all-nested-attribute';
import { OFFLINE_COLLAPSIBLE_CLASSNAME } from '../../utils/head-scripts/offline-ui/collapsible';
import { isBrowser } from '../../utils/is-browser';
import { isOfflineDocsBuild } from '../../utils/is-offline-docs-build';
import { reportAnalytics } from '../../utils/report-analytics';
import ComponentFactory from '../ComponentFactory';
import Heading from '../Heading';
import { CollapsibleNode, HeadingNode } from '../../types/ast';
import { collapsibleStyle, headerContainerStyle, headerStyle, iconStyle, innerContentStyle } from './styles';
import './styles.css';

interface CollapsibleProps {
  nodeData: CollapsibleNode;
  sectionDepth: number;
}
const Collapsible = ({ nodeData, sectionDepth, ...rest }: CollapsibleProps) => {
  const { children, options } = nodeData;
  const { id, heading, expanded, sub_heading: subHeading } = options || {};
  const { hash } = useLocation();

  // get a list of all ids in collapsible content
  // in order to set collapsible open, if any are found in url hash
  const childrenHashIds = useMemo(() => {
    return findAllNestedAttribute(children, 'id');
  }, [children]);

  const [open, setOpen] = useState(() => {
    if (isOfflineDocsBuild) return true;
    return expanded ?? true;
  });
  const headingNodeData: HeadingNode = {
    type: 'heading',
    depth: sectionDepth + 1,
    title: '',
    id: id ?? '',
    children: [{ type: 'text', value: heading ?? '' }],
    selector_ids: {},
  };

  const onIconClick = useCallback(() => {
    reportAnalytics('Click', {
      properties: {
        position: 'collapsible',
        position_context: `action: ${open ? 'collapsed' : 'expanded'}`,
        label: `heading: ${heading}`,
        label_text_displayed: `heading: ${heading}`,
        scroll_position: '0',
      },
    });
    setOpen(!open);
  }, [heading, open]);

  // open the collapsible if the hash in URL is equal to collapsible heading id,
  // or if the collapsible's children has the id
  useEffect(() => {
    if (!isBrowser) {
      return;
    }
    const hashId = hash?.slice(1) ?? '';
    if (id === hashId || childrenHashIds.includes(hashId)) {
      return setOpen(true);
    }
  }, [childrenHashIds, hash, id]);

  const rendered = useRef(false);

  // on first open, scroll the child with the URL hash id into view
  useEffect(() => {
    if (!open) return;
    if (rendered.current) return;
    rendered.current = true;
    const hashId = hash?.slice(1) ?? '';
    if (childrenHashIds.includes(hashId)) {
      const child = document?.querySelector(`#${hashId}`);
      child && child.scrollIntoView({ block: 'start', inline: 'nearest', behavior: 'smooth' });
    }
  }, [childrenHashIds, hash, open]);

  return (
    <HeadingContextProvider ignoreNextHeading={true} heading={heading}>
      <Box
        aria-expanded={open}
        className={cx('collapsible', collapsibleStyle, isOfflineDocsBuild ? OFFLINE_COLLAPSIBLE_CLASSNAME : '')}
      >
        <Box className={cx(headerContainerStyle)}>
          <Box>
            {/* Adding 1 to reflect logic in parser, but want to show up as H2 for SEO reasons */}
            <Heading className={cx(headerStyle)} sectionDepth={sectionDepth + 1} as={2} nodeData={headingNodeData}>
              {heading}
            </Heading>
            <Body baseFontSize={13}>{subHeading}</Body>
          </Box>
          <IconButton
            className={iconStyle}
            aria-labelledby={'Expand the collapsed content'}
            aria-expanded={open}
            onClick={onIconClick}
          >
            <Icon glyph={open ? 'ChevronDown' : 'ChevronRight'} />
          </IconButton>
        </Box>
        <Box className={cx(innerContentStyle)}>
          {children.map((c, i) => (
            <ComponentFactory nodeData={c} key={i} sectionDepth={sectionDepth} {...rest}></ComponentFactory>
          ))}
        </Box>
      </Box>
    </HeadingContextProvider>
  );
};
export default Collapsible;
