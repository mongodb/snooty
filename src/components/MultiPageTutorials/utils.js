import { reportAnalytics } from '../../utils/report-analytics';

export const reportMPTAnalytics = (targetSlug, variant) => {
  reportAnalytics('MultiPageTutorialNextClicked', {
    targetSlug,
    variant,
  });
};
