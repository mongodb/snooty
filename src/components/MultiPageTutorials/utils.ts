import { reportAnalytics } from '../../utils/report-analytics';

export const reportMPTAnalytics = (targetSlug: string, variant: string) => {
  reportAnalytics('MultiPageTutorialNextClicked', {
    targetSlug,
    variant,
  });
};
