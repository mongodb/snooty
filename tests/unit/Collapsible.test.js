import React from 'react';
import { render, act } from '@testing-library/react';

import { mockLocation } from '../utils/mock-location';
import Collapsible from '../../src/components/Collapsible';
import mockData from './data/Collapsible.test.json';

describe('collapsible component', () => {
  it('renders all the content in the options and children', () => {
    const renderResult = render(<Collapsible nodeData={mockData}></Collapsible>);
    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText('This is a heading')).toBeTruthy();
    expect(renderResult.getByText('This is a subheading')).toBeTruthy();
    expect(renderResult.getByText('This is collapsible content')).toBeTruthy();
  });

  it('renders the content and correct icon when interacted', async () => {
    const renderResult = render(<Collapsible nodeData={mockData}></Collapsible>);
    let button = renderResult.getByRole('button');
    const collapsedContent = renderResult.getByText('This is collapsible content');

    let icon = button.querySelector('[role=img]');
    expect(icon.getAttribute('aria-label')).toContain('Chevron');
    expect(icon.getAttribute('aria-label')).toContain('Right');
    expect(collapsedContent).not.toBeVisible();

    await act(async () => {
      button.click();
    });

    button = renderResult.getByRole('button');
    icon = button.querySelector('[role=img]');
    expect(icon.getAttribute('aria-label')).toContain('Chevron');
    expect(icon.getAttribute('aria-label')).toContain('Down');
    expect(collapsedContent).toBeVisible();
  });

  it('opens the collapsible content if hash is found in the URL', async () => {
    mockLocation(null, '/', '#this-is-a-heading');
    let renderResult = render(<Collapsible nodeData={mockData}></Collapsible>),
      button = renderResult.getByRole('button'),
      icon = button.querySelector('[role=img]');
    expect(icon.getAttribute('aria-label')).toContain('Chevron');
    expect(icon.getAttribute('aria-label')).toContain('Down');
  });
});
