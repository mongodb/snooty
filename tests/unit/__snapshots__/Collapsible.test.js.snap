// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsible component renders all the content in the options and children 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-bottom: 1px solid #E8EDEB;
}

.emotion-0:nth-child(1 of .collapsible) {
  border-top: 1px solid #E8EDEB;
  margin-top: 32px;
}

.emotion-0:nth-last-child(1 of .collapsible) {
  margin-bottom: 32px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-column-gap: 24px;
  column-gap: 24px;
  margin: 32px 0;
}

.emotion-2 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  color: #001E2B;
  margin-top: 24px;
  margin-bottom: 8px;
  color: var(--font-color-primary);
  margin-top: 0;
}

.emotion-3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 0 10px;
  visibility: hidden;
}

.emotion-4 {
  color: #889397;
  vertical-align: middle;
  margin-top: -2px;
}

.emotion-5 {
  margin-top: -175px;
  position: absolute;
  padding-bottom: 2px;
}

.emotion-7 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
}

.emotion-7 strong,
.emotion-7 b {
  font-weight: 700;
}

.emotion-8 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: color,box-shadow;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  color: #5C6C75;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #5C6C75;
}

.emotion-8:before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-8:active:before,
.emotion-8:hover:before,
.emotion-8:focus:before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-8:focus {
  outline: none;
}

.emotion-8:active,
.emotion-8:hover {
  color: #001E2B;
}

.emotion-8:active:before,
.emotion-8:hover:before {
  background-color: rgba(61,79,88,0.1);
}

.emotion-8:focus-visible {
  color: #001E2B;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-8:focus-visible:before {
  background-color: rgba(61,79,88,0.1);
}

.emotion-9 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-10 {
  overflow: hidden;
  height: 0;
  visibility: hidden;
}

.emotion-11 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  margin-bottom: 16px;
  color: var(--font-color-primary);
}

.emotion-11 strong,
.emotion-11 b {
  font-weight: 700;
}

.emotion-12 {
  display: table;
  margin: 24px 0;
  min-width: 150px;
  table-layout: fixed;
  width: 100%;
}

.emotion-12 button>div>div {
  font-size: 13px;
}

.emotion-12>div>div {
  display: grid;
  grid-template-columns: code panel;
}

.emotion-12>div {
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  display: grid;
  border-color: #E8EDEB;
}

.dark-theme .emotion-12>div {
  border-color: #3D4F58;
}

.emotion-12 pre {
  background-color: #F9FBFA;
  color: #001E2B;
}

.dark-theme .emotion-12 pre {
  background-color: #001E2B;
  color: #F9FBFA;
}

.emotion-12 [data-testid='leafygreen-code-panel'] {
  background-color: #FFFFFF;
  border-color: #E8EDEB;
}

.dark-theme .emotion-12 [data-testid='leafygreen-code-panel'] {
  background-color: #3D4F58;
  border-color: #3D4F58;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword.lg-highlight-function,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword.lg-highlight-class,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-tag,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-attr,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-pseudo,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-id,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-class {
  color: #CC3887;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword.lg-highlight-function,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-keyword.lg-highlight-class,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-tag,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-attr,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-pseudo,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-id,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-selector-class {
  color: #FF7DC3;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-regexp,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-number,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-literal,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-function.lg-highlight-title {
  color: #016ee9;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-regexp,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-number,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-literal,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-function.lg-highlight-title {
  color: #2dc4ff;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-quote,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-section,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-name {
  color: #007ab8;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-quote,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-section,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-name {
  color: #a5e3ff;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-string,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-addition {
  color: #12824D;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-string,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-addition {
  color: #35DE7B;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta-string {
  color: #956d00;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta-string {
  color: #EDB210;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-variable,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-deletion,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-symbol,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-bullet,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-link,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-attr,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-attribute,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-language,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-template-variable,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-built_in,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-type,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-params {
  color: #D83713;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-variable,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-deletion,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-symbol,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-bullet,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-meta,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-link,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-attr,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-attribute,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-language,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-template-variable,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-built_in,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-type,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-params {
  color: #FF6F44;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-title,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-class.lg-highlight-title {
  color: #001E2B;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-title,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-class.lg-highlight-title {
  color: #F9FBFA;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-doctag,
.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-formula {
  color: #001E2B;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-doctag,
.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-formula {
  color: #F9FBFA;
}

.emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-comment {
  color: #3D4F58;
}

.dark-theme .emotion-12 [class*="lg-highlight-hljs-"] .lg-highlight-comment {
  color: #C1C7C6;
}

.emotion-13 {
  border: 1px solid #E8EDEB;
  border-radius: 12px;
  overflow: hidden;
}

.emotion-14 {
  position: relative;
  display: grid;
  grid-template-areas: 'code panel';
  grid-template-columns: auto 38px;
  border-radius: inherit;
  z-index: 0;
}

.emotion-14:before,
.emotion-14:after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  height: 100%;
  width: 40px;
  border-radius: 40%;
  box-shadow: unset;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
}

.emotion-14:before {
  grid-column: 1;
  left: -40px;
}

.emotion-14:after {
  grid-column: 2;
}

.emotion-15 {
  grid-area: code;
  overflow-x: auto;
  border-radius: inherit;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
  position: relative;
  -webkit-transition: box-shadow 100ms ease-in-out;
  transition: box-shadow 100ms ease-in-out;
  white-space: pre;
  background-color: #F9FBFA;
  color: #001E2B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 6px;
  padding-bottom: 6px;
}

@media only screen and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-15 {
    white-space: pre-wrap;
  }
}

@media only screen and (min-device-width: 813px) and (-webkit-min-device-pixel-ratio: 2) {
  .emotion-15 {
    white-space: pre;
  }
}

.emotion-15:focus-visible {
  outline: none;
  box-shadow: 0 0 0 2px #0498EC inset;
}

.emotion-16 {
  color: inherit;
  font-family: 'Source Code Pro',Menlo,monospace;
  font-size: 13px;
  line-height: 20px;
}

.emotion-17 {
  border-spacing: 0;
}

.emotion-18 {
  border-spacing: 0;
  vertical-align: top;
  padding: 0 16px;
}

.emotion-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 4px;
  background-color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
  border-left: solid 1px;
  border-color: #E8EDEB;
  z-index: 2;
  grid-area: panel;
}

.emotion-19 svg {
  width: 16px;
  height: 16px;
}

<div
    class="collapsible emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div>
        <h2
          class="contains-headerlink emotion-2"
          weight="medium"
        >
          This is a heading
          <a
            class="headerlink emotion-3"
            href="#this-is-a-heading"
            title="Permalink to this heading"
          >
            <svg
              aria-label="Link Icon"
              class="emotion-4"
              fill="none"
              height="12"
              role="img"
              viewBox="0 0 16 16"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.039 9.953a3.596 3.596 0 0 1-.327-.38l1.45-1.445a1.553 1.553 0 0 0 2.496.423l2.885-2.87a1.57 1.57 0 0 0 .01-2.213 1.553 1.553 0 0 0-2.203-.01l-.379.377A.995.995 0 0 1 8.56 3.83a1.005 1.005 0 0 1 .006-1.418l.38-.377a3.542 3.542 0 0 1 5.024.023 3.58 3.58 0 0 1-.022 5.047l-2.884 2.871a3.542 3.542 0 0 1-5.025-.022Z"
                fill="currentColor"
              />
              <path
                d="M9.961 6.047c.12.12.228.248.327.38l-1.45 1.445a1.553 1.553 0 0 0-2.496-.423l-2.885 2.87a1.57 1.57 0 0 0-.01 2.213 1.553 1.553 0 0 0 2.203.01l.379-.377a.995.995 0 0 1 1.411.006 1.005 1.005 0 0 1-.006 1.418l-.38.377a3.542 3.542 0 0 1-5.024-.023 3.58 3.58 0 0 1 .022-5.047l2.884-2.871a3.542 3.542 0 0 1 5.025.022Z"
                fill="currentColor"
              />
            </svg>
          </a>
          <div
            class="emotion-5 emotion-6"
            id="this-is-a-heading"
          />
        </h2>
        <p
          class="emotion-7"
        >
          This is a subheading
        </p>
      </div>
      <button
        aria-disabled="false"
        aria-labelledby="Expand the collapsed content"
        class="emotion-8"
        tabindex="0"
      >
        <div
          class="emotion-9"
        >
          <svg
            aria-label="Chevron Right Icon"
            class=""
            fill="none"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.364 14.364a1 1 0 0 0 1.414 0l4.95-4.95.707-.707a1 1 0 0 0 0-1.414l-.707-.707-4.95-4.95a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 0 1.414L8.899 8l-4.242 4.243a1 1 0 0 0 0 1.414l.707.707Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="emotion-10"
    >
      <p
        class="emotion-11"
      >
        This is collapsible content
      </p>
      <div
        class="emotion-12"
      >
        <div
          class="emotion-13"
        >
          <div
            class="emotion-14"
          >
            <pre
              class="emotion-15"
              tabindex="-1"
            >
              <code
                class="lg-highlight-hljs-light javascript emotion-16"
              >
                <table
                  class="emotion-17"
                >
                  <tbody>
                    <tr
                      class=""
                    >
                      <td
                        class="emotion-18"
                      >
                        <span
                          class="lg-highlight-title lg-highlight-class"
                        >
                          This
                        </span>
                         is code within collapsible content
                      </td>
                    </tr>
                  </tbody>
                </table>
              </code>
            </pre>
            <div
              class="emotion-19"
              data-testid="leafygreen-code-panel"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
